ui-v2/src/components/work-pools/work-pool-queue-page-header/work-pool-queue-page-header.test.tsx

```diff
@@ -0,0 +1,191 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { render, screen } from "@testing-library/react";
+import { describe, expect, it, vi } from "vitest";
+import type { WorkPoolQueue } from "@/api/work-pool-queues";
+import { TooltipProvider } from "@/components/ui/tooltip";
+import { createFakeWorkPoolQueue } from "@/mocks";
+import { WorkPoolQueuePageHeader } from "./work-pool-queue-page-header";
+
+// Mock Tanstack Router
+vi.mock("@tanstack/react-router", async () => {
+	const actual = await vi.importActual("@tanstack/react-router");
+	return {
+		...actual,
+		Link: ({
+			children,
+			to,
+			params,
+		}: {
+			children: React.ReactNode;
+			to: string;
+			params?: Record<string, string>;
+		}) => {
+			const href = params
+				? to.replace("$workPoolName", params.workPoolName || "")
+				: to;
+			return <a href={href}>{children}</a>;
+		},
+		useNavigate: () => vi.fn(),
+		createLink:
+			() =>
+			({
+				children,
+				to,
+				params,
+			}: {
+				children: React.ReactNode;
+				to: string;
+				params?: Record<string, string>;
+			}) => {
+				const href = params
+					? to.replace("$workPoolName", params.workPoolName || "")
+					: to;
+				return <a href={href}>{children}</a>;
+			},
+	};
+});
+
+// Mock the sub-components
+vi.mock("@/components/work-pools/work-pool-queue-toggle", () => ({
+	WorkPoolQueueToggle: ({ queue }: { queue: WorkPoolQueue }) => (
+		<div data-testid="work-pool-queue-toggle">Toggle for {queue.name}</div>
+	),
+}));
+
+vi.mock("@/components/work-pools/work-pool-queue-menu", () => ({
+	WorkPoolQueueMenu: ({ queue }: { queue: WorkPoolQueue }) => (
+		<div data-testid="work-pool-queue-menu">Menu for {queue.name}</div>
+	),
+}));
```

Comment:
This is too many mocks that are likely unnecessary. Follow the establish pattern for handling a router depedency when unit testing components.